### PR TITLE
@grafana/ui: When to use Checkbox docs

### PR DESCRIPTION
**Pull request by jessover9000**
This is a docs update. Added info about how checkboxes generally work and when to use another component instead.

### CODE REVIEW

File: packages/grafana-ui/src/components/Forms/Checkbox.mdx [hunk 1]
```diff
@@ -5,6 +5,12 @@ import { Checkbox } from './Checkbox';
 
 # Checkbox
 
+### When to use
+
+`Checkbox` can be used in groups, where the single checkboxes have no dependencies. That means that selecting one doesn’t affect any other `Checkbox`. Checked represents true, un-checked represent false. So you can use them to select a binary option or multiple options in a set. When adding a description to your `Checkbox`, use positive commands, not negatives like "do not …".
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/Forms/Checkbox.mdx]: `can` or `should` be used in group?

File: packages/grafana-ui/src/components/Forms/Checkbox.mdx [hunk 2]
```diff
@@ -5,6 +5,12 @@ import { Checkbox } from './Checkbox';
...
+
+Checkboxes typically only trigger changes after sending a form. If your component should trigger a change immediately, it's better to use a toggle switch instead. Furthermore, checkboxes are not mutually exclusive. If you want to offer a mutually exclusive choice, use `RadioButtonGroup` or a `Select` dropdown.
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/Forms/Checkbox.mdx]: Maybe use other wording than mutually exclusive or elaborate a bit on that.

**Review by tskarhed — COMMENTED** [review]
I get the feeling that checkboxes are used exclusively in submittable forms. Is that correct?

File: packages/grafana-ui/src/components/Forms/Checkbox.mdx (hunk 1 above)
Comment (sarlinska) [reply packages/grafana-ui/src/components/Forms/Checkbox.mdx:10]: 1. I would start with:
"Checked represents true, un-checked represent false."
And then go about using them in groups.
2. Examples of positive commands would be useful – it's hard to imagine how they could sound. Show examples of do's and don'ts.

File: packages/grafana-ui/src/components/Forms/Checkbox.mdx (hunk 2 above)
Comment (sarlinska) [reply packages/grafana-ui/src/components/Forms/Checkbox.mdx:12]: Again examples could be useful.

**Review by sarlinska — COMMENTED** [review]
Just a couple of comments, mostly about adding some examples to better depict what we mean.

File: packages/grafana-ui/src/components/Forms/Checkbox.mdx [hunk 3]
```diff
@@ -5,6 +5,27 @@ import { Checkbox } from "./Checkbox";
...
+**DO:** [ ] Hide options
+**DON'T:** [ ] Do not show options
+
+Checkboxes typically only trigger changes after sending a form. If your component should trigger a change immediately, it's better to use a toggle switch instead. Furthermore, checkboxes are not mutually exclusive. That means that selecting one will not disable the others or impact them in any other way. If you want to offer a mutually exclusive choice, use `RadioButtonGroup` or a `Select` dropdown.
+
+**DO:**
+Show legends
+[ ] Blue
+[ ] Green
+[ ] Red
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Forms/Checkbox.mdx:21]: this example is weird to me:)

Comment (dprokop) [reply packages/grafana-ui/src/components/Forms/Checkbox.mdx:21]:
```suggestion
Show series
[ ] A-series
[ ] B-series
[ ] C-series
```

File: packages/grafana-ui/src/components/Forms/Checkbox.mdx [hunk 4]
```diff
@@ -5,6 +5,27 @@ import { Checkbox } from "./Checkbox";
...
+
+**DON'T:**
+Show only
+[ ] Blue
+[ ] Green
+[ ] Red
```
Comment (dprokop) [inline packages/grafana-ui/src/components/Forms/Checkbox.mdx:27]:
```suggestion
Show only
[ ] A-series
[ ] B-series
[ ] C-series
```

**Review by dprokop — APPROVED** [review]

**Pull request merged**